src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/study/StudyAdherenceReportGenerator.java

```diff
@@ -135,11 +110,48 @@ public StudyAdherenceReport generate(AdherenceState state) {
             days.addAll(entry.getValue());
         };
 
+        // Some study-wide calculations:
+        
+        // Calculate the date range, but make sure it includes the study start date
+        // so the user can see if the participant is acting well before the study start 
+        // date or well after, etc.
+        DateRange dateRange = null;
+        if (eventReport.getDateRangeOfAllStreams() != null) {
+            DateRange streamDates = eventReport.getDateRangeOfAllStreams();
+            LocalDate startDate = streamDates.getStartDate();
+            LocalDate endDate = streamDates.getEndDate();
+            if (studyStartDate != null) {
+                if (studyStartDate.isBefore(startDate)) {
+                    startDate = studyStartDate;
+                } else if (studyStartDate.isAfter(endDate)) {
+                    endDate = studyStartDate;
+                }
+            }
+            if (startDate.isEqual(endDate) || startDate.isBefore(endDate)) {
+                dateRange = new DateRange(startDate, endDate);
+            }
+        }
+        
+        // Study-wide progress and adherence. Don't calculate adherence if there's no schedule.
+        ParticipantStudyProgress progression = calculateProgress(state, ImmutableList.of(studyStream));
+        Integer adherence = null;
+        if (!ParticipantStudyProgress.NO_ADHERENCE.contains(progression)) {
+            adherence = calculateAdherencePercentage(ImmutableList.of(studyStream));
+        }
+        // If the earliest date is before the study start date, we're going to offset the weekInStudy 
+        // field to show there are scheduled activities occurring earlier in the study design.
+        int weekOffset = 0;
+        if (earliestDate != null && studyStartDate != null && earliestDate.isBefore(studyStartDate)) {
+            int days = Days.daysBetween(earliestDate, studyStartDate).getDays();
+            weekOffset = days/7;
+        }
```

Comment:
Off-by-one error: take the example of a day or two before the study, this this is Week 0, is it not?

Reply:
It is actually okay as the earliestDate and studyStartDate are in the same week. It just changes where the two are in relation to one another in the week. It's only once the offset is 7+ that we need to create "negative" weeks (0, -1, etc.).